spec/e2e/support/target-blank.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @typedef SimplifiedLink
+ *
+ * @prop {string=} innerText
+ * @prop {string=} href
+ * @prop {string=} className
+ * @prop {string=} target
+ */
+
+/** @type {import('puppeteer').Page} */
+/** @return {Promise<SimplifiedLink[]>} */
+async function getLinks(page) {
+  return await page.evaluate(() =>
+    Array.prototype.map.call(document.querySelectorAll('a'), (a) => {
```

Comment:
No strong preference (`map.call` has some upside of skipping conversion to array), but possibly more readable as...

```suggestion
    Array.from(document.querySelectorAll('a')).map((a) => {
```

Or:

```suggestion
    [...document.querySelectorAll('a')].map((a) => {
```

Reply:
I tried `Array.from` and got an undefined method error... but I'll try again

Reply:
Ok `Array.from` did work this time 🤷 ... going to see if I can combine this with the `$$eval` suggestion below